src/components/toasts/WalletConnectToast.tsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+import { toast } from 'sonner-native';
+import { globalColors } from '@/design-system';
+import { typeHierarchy } from '@/design-system/typography/typeHierarchy';
+import { IS_IOS } from '@/env';
+import * as i18n from '@/languages';
+import { fontWithWidth } from '@/styles';
+import font from '@/styles/fonts';
+import { THICK_BORDER_WIDTH } from '@/__swaps__/screens/Swap/constants';
+import { opacity } from '@/__swaps__/utils/swaps';
+import { isDarkTheme } from '@/theme/ThemeContext';
+
+const WALLETCONNECT_TOAST_ID = 'walletConnect';
+const HIDE_BUFFER_MS = 750;
+
+let lastHideTime = 0;
+
+export const hideWalletConnectToast = () => {
+  lastHideTime = Date.now();
+  toast.dismiss(WALLETCONNECT_TOAST_ID);
+};
+
+export const showWalletConnectToast = async ({ isTransactionRequest = false }: { isTransactionRequest?: boolean } = {}) => {
+  if (Date.now() - lastHideTime <= HIDE_BUFFER_MS) {
+    // If the transaction request loads and hide is called before the toast is shown, skip showing it
+    return;
+  }
+
+  const isDarkMode = await isDarkTheme();
```

Comment:
why this instead of [useColorMode](https://github.com/rainbow-me/rainbow/blob/eac19c04b756d70a84197fbd6ad313c4422444cf/src/design-system/color/ColorMode.tsx#L34)?